src/extension.ts

```diff
@@ -199,28 +199,51 @@ async function startClients(folders = prunedFolderNames()) {
 	console.log('Starting:', folders);
 	const command = await pathToBinary();
 	const disposables: vscode.Disposable[] = [];
-	for (const folder of folders) {
-		if (!clients.has(folder)) {
-			const commandPrefix = shortUid.seq();
-			const client = newClient(command, folder, commandPrefix);
-			client.onReady().then(() => {
-				reporter.sendTelemetryEvent('startClient');
-			});
-			client.onDidChangeState((event) => {
-				if (event.newState === ClientState.Stopped) {
-					clients.delete(folder);
-					reporter.sendTelemetryEvent('stopClient');
-				}
-			});
-			disposables.push(client.start());
-			clients.set(folder, { commandPrefix, client });
-		} else {
-			console.log(`Client for folder: ${folder} already started`);
+
+	const firstClient = configureClient(command, folders[0])
```

Comment:
Not sure `firstClient` is the best name for what this is, but I couldn't come up with a better one :|

Reply:
How about `primaryClient`?

Reply:
Yeah, that's better, although I was more referring to the fact that the variable contains the commandPrefix + the client instance.

Reply:
I agree "client" is getting a little overloaded – one direction you could go is to use `client`, `clientWithPrefix`, `clientWithConfig`, etc.